websites/K/KOLEO/presence.ts

```diff
@@ -0,0 +1,978 @@
+	presenceData.details = "Nie znaleziono strony.";
+	presenceData.largeImageKey = Assets.Logo;
+	presenceData.state = "";
+	presenceData.smallImageText = "Zgubił się...";
+	presenceData.smallImageKey = Assets.Question;
+}
+
+const operators = [
+	"pkp-intercity",
+	"polregio",
+	"arriva",
+	"leo-express",
+	"koleje-wielkopolskie",
+	"koleje-dolnoslaskie",
+	"koleje-mazowieckie",
+	"koleje-malopolskie",
+	"wkd",
+	"lka",
+	"koleje-slaskie",
+	"skm-trojmiasto",
+];
+
+presence.on("UpdateData", async () => {
+	const { href } = document.location,
+		presenceData: PresenceData = {
+			details: "Ładowanie.",
+			largeImageKey: Assets.Logo,
+			smallImageText: "Ładowanie...",
+			smallImageKey: Assets.Train,
+			startTimestamp: browsingTimestamp,
+		},
+		privacySetting = await presence.getSetting<boolean>("privacy");
+
+	let currentPath;
+
+	if (href.startsWith("https://koleo.pl")) {
+		currentPath = href.replace("https://koleo.pl", "");
+		presenceData.details = "Szuka połączenia rozkładu koleji.";
+		if (
+			href.endsWith("koleo.pl/") ||
+			href.endsWith("koleo.pl/#") ||
+			currentPath.includes("rozklad-jazdy") ||
+			operators.includes(currentPath.split("/")[1])
+		) {
+			let startText: string, endText: string, dateText: string;
+
+			if (href.endsWith("koleo.pl/") || href.endsWith("koleo.pl/#")) {
+				const startStation = document.querySelector<HTMLInputElement>(
+						"#query_start_station"
+					),
+					endStation = document.querySelector(
+						"#query_end_station"
+					) as HTMLInputElement,
+					date = document.querySelector("#query_date") as HTMLInputElement;
+				if (startStation && endStation && date) {
+					startText = startStation.value;
+					endText = endStation.value;
+					dateText = date.value;
+					presenceData.state = "W całym KOLEO...";
+				}
+			} else if (
+				currentPath.includes("rozklad-jazdy") ||
+				operators.includes(currentPath.split("/")[1])
+			) {
+				const startStationButton = document.querySelector(".closest-station"),
+					endStationButton = document.querySelector(".swap-stations"),
+					dateInputWrapper = document.querySelector(
+						".form-date-input__input-wrapper"
+					);
+
+				if (startStationButton && endStationButton && dateInputWrapper) {
+					startText = (
+						startStationButton
+							.closest(".icon-input-action")
+							.querySelector("input") as HTMLInputElement
+					).value;
+					endText = (
+						endStationButton
+							.closest(".icon-input-action")
+							.querySelector("input") as HTMLInputElement
+					).value;
+					dateText = (
+						dateInputWrapper.querySelector("input") as HTMLInputElement
+					).value;
+
+					presenceData.state = `W ${
+						(
+							document.querySelector(".top-banner__heading") || {
+								textContent: "caŁym KOLEO",
+							}
+						).textContent
+					}...`;
+				}
+			}
+			presenceData.smallImageText = "Szuka połączenia...";
+			presenceData.buttons = [
+				{ label: "Rozkład jazdy", url: "https://koleo.pl/rozklad-jazdy" },
```

Comment:
Button labels should be capitalized
```ts
{ label: "Rozkład Jazdy", url: "https://koleo.pl/rozklad-jazdy" },
```

Reply:
Both of the letters?!?!?

Reply:
and also why

Reply:
It's the button style this repository tries to use.

Reply:
In my other presences i didnt have to do this but, alright..